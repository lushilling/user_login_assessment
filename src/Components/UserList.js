import React from 'react';
import { Table } from "reactstrap";
import User from './User';

export default class UserList extends React.Component {

    render() {
        return (
            <div>
                <Table>
                    <tbody>
                        <tr>{this.props.data.map((user, username) => (
                                 <User key={username} onLoadFunction={this.props.onLoadFunction} username={user.username} email={user.email} />       
                        ))}</tr>
                    </tbody>
                </Table>
            </div>

        );
    }
}