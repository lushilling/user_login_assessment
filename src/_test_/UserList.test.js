import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import UserList from '../Components/UserList';


it('renders userList', () => {
    const tree = renderer.create(<UserList />).toJSON()
    expect(tree).toMatchSnapshot();
})

it('renders userlist without crashing', () => {
    shallow(<UserList />);
});