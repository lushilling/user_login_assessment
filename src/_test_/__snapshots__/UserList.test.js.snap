// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders userList 1`] = `
<div>
  <form
    className="form-inline"
    onSubmit={[Function]}
  >
    <div
      className="mb-2 mr-sm-2 mb-sm-0 form-group"
    >
      <label
        className="mr-sm-2"
        htmlFor="username"
      >
        Username: 
      </label>
      <input
        className="form-control"
        id="username"
        name="username"
        placeholder="Enter Username"
        type="text"
      />
    </div>
    <div
      className="mb-2 mr-sm-2 mb-sm-0 form-group"
    >
      <label
        className="mr-sm-2"
        htmlFor="Email"
      >
        Email: 
      </label>
      <input
        className="form-control"
        id="Email"
        name="Email"
        placeholder="Enter Email"
        type="Email"
      />
    </div>
    <div
      className="mb-2 mr-sm-2 mb-sm-0 form-group"
    >
      <label
        className="mr-sm-2"
        htmlFor="password"
      >
        Password: 
      </label>
      <input
        className="form-control"
        id="password"
        name="password"
        placeholder="Enter Password"
        type="password"
      />
    </div>
    <div
      className="mb-2 mr-sm-2 mb-sm-0 form-group"
    >
      <label
        className="mr-sm-2"
        htmlFor="password"
      >
        Re-Enter Password: 
      </label>
      <input
        className="form-control"
        id="password"
        name="password"
        placeholder="Enter Password"
        type="password"
      />
    </div>
    <button
      aria-label={null}
      className="btn btn-secondary"
      onClick={[Function]}
    >
      Login
    </button>
  </form>
  <p
    style={
      Object {
        "color": "red",
      }
    }
  />
  <p
    style={
      Object {
        "color": "red",
      }
    }
  />
</div>
`;
